File: 242-valid-anagram/242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {

        Deque<Character> stack = new LinkedList<>();
        if(s.length() != t.length()) return false;

        for(int i = 0, j = 0; i < s.length(); i++) {
            stack.push(s.charAt(i));
            while(!stack.isEmpty() && stack.contains(t.charAt(j))) {
                stack.remove(t.charAt(j));
                j++;
            }
        }
        return stack.isEmpty();
    
    }
}

File: 242-valid-anagram/242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        char = [0 for _ in range(26)]

        for c in s:
            char[ord(c)-97] += 1
        for c in t:
            char[ord(c)-97] -= 1

        for n in char:
            if n != 0:
                return False

        return True